saleor/static/dashboard-next/orders/views/OrderDetails/OrderDetailsMessages.tsx

```diff
@@ -24,309 +24,291 @@ import { OrderVoid } from "../../types/OrderVoid";
+    }
+  };
+  const handleShippingMethodUpdate = (data: OrderShippingMethodUpdate) => {
+    if (!maybe(() => data.orderUpdateShipping.errors.length)) {
+      pushMessage({
+        text: i18n.t("Shipping method successfully updated", {
+          context: "notification"
+        })
+      });
+    } else {
+      pushMessage({
+        text: i18n.t("Could not update shipping method", {
+          context: "notification"
+        })
+      });
+    }
+    navigate(orderUrl(data.orderUpdateShipping.order.id), true);
+  };
+  const handleOrderLineDelete = (data: OrderLineDelete) => {
+    if (!maybe(() => data.draftOrderLineDelete.errors.length)) {
+      pushMessage({
+        text: i18n.t("Order line deleted", {
+          context: "notification"
+        })
+      });
+    } else {
+      pushMessage({
+        text: i18n.t("Could not delete order line", {
+          context: "notification"
+        })
+      });
+    }
+  };
+  const handleOrderLineAdd = (data: OrderLineAdd) => {
+    if (!maybe(() => data.draftOrderLinesCreate.errors.length)) {
+      pushMessage({
+        text: i18n.t("Order line added", {
+          context: "notification"
+        })
+      });
+      navigate(orderUrl(data.draftOrderLinesCreate.order.id), true);
+    } else {
+      pushMessage({
+        text: i18n.t("Could not create order line", {
+          context: "notification"
+        })
+      });
+    }
+  };
+  const handleOrderLineUpdate = (data: OrderLineUpdate) => {
+    if (!maybe(() => data.draftOrderLineUpdate.errors.length)) {
+      pushMessage({
+        text: i18n.t("Order line updated", {
+          context: "notification"
+        })
+      });
+    } else {
+      pushMessage({
+        text: i18n.t("Could not update order line", {
+          context: "notification"
+        })
+      });
+    }
+  };
+  const handleOrderFulfillmentCancel = (data: OrderFulfillmentCancel) => {
+    if (!maybe(() => data.orderFulfillmentCancel.errors.length)) {
+      pushMessage({
+        text: i18n.t("Fulfillment successfully cancelled", {
+          context: "notification"
+        })
+      });
+      navigate(orderUrl(data.orderFulfillmentCancel.order.id), true);
+    } else {
+      pushMessage({
+        text: i18n.t("Could not cancel fulfillment", {
+          context: "notification"
+        })
+      });
+    }
+  };
+  const handleOrderFulfillmentUpdate = (
+    data: OrderFulfillmentUpdateTracking
+  ) => {
+    if (!maybe(() => data.orderFulfillmentUpdateTracking.errors.length)) {
+      pushMessage({
+        text: i18n.t("Fulfillment successfully updated", {
+          context: "notification"
+        })
+      });
+      navigate(orderUrl(data.orderFulfillmentUpdateTracking.order.id), true);
+    } else {
+      pushMessage({
+        text: i18n.t("Could not update fulfillment", {
+          context: "notification"
+        })
+      });
+    }
+  };
+  const handleDraftFinalize = (data: OrderDraftFinalize) => {
```

Comment:
I think at least half of these could reuse functionality, i.e. accept result from `OrderDraftFinalize` and others which would be `{errors, order}` and a message. 🙄

Reply:
There are plans for some refactoring here, but not in this PR.